src/Bicep.Core/Analyzers/Linter/LinterRulesProvider.cs

```diff
@@ -22,6 +23,7 @@ public LinterRulesProvider()
             this.linterRulesLazy = new Lazy<ImmutableDictionary<string, string>>(() => GetLinterRulesInternal().ToImmutableDictionary());
         }
 
+        [UnconditionalSuppressMessage("Trimming", "IL2072:Target parameter argument does not satisfy 'DynamicallyAccessedMembersAttribute' in call to target method. The return value of the source method does not have matching annotations.", Justification = "List of types comes from a source analyzer")]
```

Comment:
If the suppression is unavoidable, feels like it would be worth scoping it down s.t. only one is necessary, and we encapsulate the 'unsafe' logic in a single minimal function s.t. the scope of the suppression is obvious.

Reply:
This is the same pattern as the previous comment, so hopefully my source gen change will eliminate it too.